src/main/java/underlay/Underlay.java

```diff
@@ -0,0 +1,168 @@
+package underlay;
+
+import blockchain.LightChainRMIInterface;
+import java.io.FileNotFoundException;
+import java.rmi.Naming;
+import java.rmi.RemoteException;
+import org.apache.log4j.Logger;
+import skipGraph.NodeInfo;
+import skipGraph.RMIInterface;
+import underlay.requests.*;
+import underlay.responses.*;
+import util.Util;
+
+
+
+/**
+ * The underlay class is used to abstract away the RMI primitives used in the skipGraph and
+ * lightchain nodes.
+ */
+public class Underlay {
+
+  /** The underlying RMIInterface instance. Is used for skipnode calls */
+  private RMIInterface targetRMI;
+
+  /** The underlying LightChainRMIInterface instance. Is used for lightchainnode class */
+  private LightChainRMIInterface targetLightChainRMI;
+
+  private final Logger logger = Logger.getLogger("");
```

Comment:
These all should be set as part of the `Underlay`'s constructor. As a general best practice, we should not have separate setters for member fields that we do not expect to change and be updated during the runtime. Please also remove the setters.

Reply:
These (targetRMI and targetLightChainRMI) are set each time sendMessage is called - they change based on address, so I'm not sure if what you've said applies here.

Reply:
Right, thanks.